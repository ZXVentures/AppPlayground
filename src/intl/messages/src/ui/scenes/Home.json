[
  {
    "id": "home.welcome",
    "defaultMessage": "Welcome to AppPlayground!"
  },
  {
    "id": "home.intro",
    "defaultMessage": "This React Native Boilerplate provides Redux, GraphQL through Apollo, Navigation, and a Test Suite."
  },
  {
    "id": "home.debugInstructions",
    "defaultMessage": "Double tap R on your keyboard to reload,\n\n          Shake or press menu button for dev menu"
  },
  {
    "id": "home.modal",
    "defaultMessage": "Modal"
  }
]