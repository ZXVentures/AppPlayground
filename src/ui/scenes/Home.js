/* @flow */

import React, { Component } from 'react' // eslint-disable-line no-unused-vars
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity
} from 'react-native'

export default class Home extends Component {

  render(): React$Element<any> {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to AppPlayground!
        </Text>
        <Text style={styles.body}>
          This React Native Boilerplate provides Redux, GraphQL through Apollo, Navigation, and a Test Suite.
        </Text>
        <Text style={styles.body}>
          Double tap R on your keyboard to reload,{'\n'}
          Shake or press menu button for dev menu
        </Text>
        <View style={styles.buttonContainer}>
          <View style={styles.buttonRow}>
            <TouchableOpacity onPress={this.props.showModal}>
              <Text style={styles.button}>Modal</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  welcome: {
    fontSize: 24,
    textAlign: 'center',
    margin: 10,
    marginBottom: 20
  },
  body: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  },
  buttonContainer: {
    height: 50,
    marginTop: 20
  },
  buttonRow: {
    flex: 1,
    flexDirection: 'row'
  },
  button: {
    fontWeight: '500',
    fontSize: 18,
    marginHorizontal: 10
  }
})
