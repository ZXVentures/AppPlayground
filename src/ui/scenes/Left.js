import React, { Component } from 'react' // eslint-disable-line no-unused-vars
import {
  StyleSheet,
  View
} from 'react-native'

export default class Left extends Component {

  render(): React$Element<any> {
    return (
      <View style={styles.container}></View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#c51162'
  }
})
