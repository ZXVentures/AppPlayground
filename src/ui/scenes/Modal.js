import React, { Component } from 'react' // eslint-disable-line no-unused-vars
import {
  StyleSheet,
  View
} from 'react-native'

export default class Modal extends Component {

  render(): React$Element<any> {
    return (
      <View style={styles.container}></View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#0091ea'
  }
})
