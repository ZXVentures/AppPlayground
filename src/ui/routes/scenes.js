import React from 'react' // eslint-disable-line no-unused-vars

import Home from '../scenes/Home'
import Left from '../scenes/Left'
import Modal from '../scenes/Modal'
import Right from '../scenes/Right'

import {
  HOME,
  RIGHT,
  MODAL_TITLE,
  LEFT
} from './constants'

export default {
  [HOME]: <Home />,
  [RIGHT]: <Right />,
  [MODAL_TITLE]: <Modal />,
  [LEFT]: <Left />
}
