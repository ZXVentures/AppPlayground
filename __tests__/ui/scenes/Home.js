import 'react-native'
import React from 'react' // eslint-disable-line no-unused-vars
import { Home, mapDispatchToProps } from 'ui/scenes/Home'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('ui/scenes/Home', () => {
  it('snapshot test', () => {
    const tree = renderer.create(
      <Home />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('maps dispatch to props', () => {
    const mockDispatch = jest.fn()

    const props = mapDispatchToProps(mockDispatch)

    expect(props).not.toBeNull()
    expect(Object.keys(props).length).toBe(1) // you need to add a test if this fails
    props.showModal()
    expect(mockDispatch).toHaveBeenCalled()
  })
})
