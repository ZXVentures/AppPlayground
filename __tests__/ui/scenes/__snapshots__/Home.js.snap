exports[`ui/scenes/Home snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
        "margin": 10,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }>
    Welcome to AppPlayground!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    This React Native Boilerplate provides Redux, GraphQL through Apollo, Navigation, and a Test Suite.
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
    Double tap R on your keyboard to reload,
    
    
    Shake or press menu button for dev menu
  </Text>
  <View
    style={
      Object {
        "height": 50,
        "marginTop": 20,
      }
    }>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "500",
              "marginHorizontal": 10,
            }
          }>
          Modal
        </Text>
      </View>
    </View>
  </View>
</View>
`;
